src/partition/repart.c

```diff
@@ -4775,6 +4775,18 @@ static int make_subvolumes_set(
         return 0;
 }
 
+static usec_t epoch_or_utime_omit(void) {
+        uint64_t epoch;
+
+        if (secure_getenv_uint64("SOURCE_DATE_EPOCH", &epoch) >= 0) {
+                if (epoch > UINT64_MAX / USEC_PER_SEC) /* Overflow check */
+                        return USEC_INFINITY;
+                return (usec_t) epoch * USEC_PER_SEC;
+        }
+
+        return (usec_t) UTIME_OMIT;
```

Comment:
uh. UTIME_OMIT is a special value you assign to the second-granularity .tv_sec or the nano-second granularity .tv_nsec fields.
 

Here, we return a micro-second granularity usec_t. It feels a bit weird returning UTIME_OMIT here.

Maybe return USEC_MAX here instead, and make sure to convert that to UTIME_OMIT when converting from usec_t to timespec